server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts

```diff
@@ -91,31 +91,40 @@ ContactedPeopleRoute.get('/allContacts/:investigationId', (request: Request, res
         })
 });
 
-ContactedPeopleRoute.post('/interactedContacts', (request: Request, response: Response) => {
+ContactedPeopleRoute.post('/interactedContacts',  (request: Request, response: Response) => {
+    const requestData = request.body;
+
     const epidemiologyNumber = +response.locals.epidemiologynumber;
-    const isThereDoneContact = request.body.unSavedContacts.contacts.some((contact : InteractedContact) => contact.contactStatus === DONE_CONTACT);
-    const isSingleContact = request.body.unSavedContacts?.contacts.length === 1;
+    const isThereDoneContact = requestData.unSavedContacts.contacts.some((contact : InteractedContact) => contact.contactStatus === DONE_CONTACT);
+    const isSingleContact = requestData.unSavedContacts?.contacts.length === 1;
+
     const workflow = `Saving ${isSingleContact ? 'single': 'all'} contact${isSingleContact ? '' : 's'}`;
     const interactedContactsLogger = logger.setup({
         workflow: `Saving ${isSingleContact ? 'single': 'all'} contact${isSingleContact ? '' : 's'}`,
         investigation: response.locals.epidemiologynumber,
         user: response.locals.user.id
     });
-    interactedContactsLogger.info(`launching graphql API request with parameter: ${JSON.stringify(request.body)}`, Severity.LOW);
-    graphqlRequest(UPDATE_LIST_OF_CONTACTS, response.locals, { unSavedContacts: JSON.stringify(request.body)})
-        .then((result: any) => {
-            if(result.data.updateContactPersons) {
-                interactedContactsLogger.info('got response from DB', Severity.LOW);
-                isThereDoneContact && sendSavedInvestigationToIntegration(epidemiologyNumber, workflow, response.locals.user.id);
-                response.send(result);
-            } else if(result.errors) {
-                handleDBErrors(response, result.errors[0].message, workflow);
-            }
-        })
-        .catch(error => {
-            interactedContactsLogger.error(`got error from DB from graphql API: ${error}`, Severity.HIGH);
-            response.status(errorStatusCode).json({error: 'failed to save all the contacts'});
-        })
+
+    interactedContactsLogger.info(`launching the graphql API request for address creation with the parameters: ${JSON.stringify(requestData)}`,
+        Severity.LOW);
+    try {
+        interactedContactsLogger.info(`launching graphql API request with parameter: ${JSON.stringify(requestData)}`, Severity.LOW);
+         graphqlRequest(UPDATE_LIST_OF_CONTACTS, response.locals, { unSavedContacts: JSON.stringify(requestData)})
+            .then((result: any) => {
+                if (result.data.updateContactPersons) {
```

Comment:
result.data should be checked for undefined

Reply:
rebase deleted this code